webapp/tests/data_access/test_form_data_controller.py

```diff
@@ -0,0 +1,34 @@
+import unittest
+from unittest.mock import patch
+
+import fakeredis as fakeredis
+
+from pydantic import MissingError
+from app.data_access.form_data_controller import save, get
+from app.config import Config
+
+redis_fake = fakeredis.FakeStrictRedis()
+
+
+@patch("app.data_access.form_data_controller.r", redis_fake)
+class TestFormDataController(unittest.TestCase):
```

Comment:
Personally, I would pair one test class with one method/function to keep it structured even if more tests are added later